Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -77,6 +76,8 @@ void UPSHUDComponent::OnUnregister()
 		FWidgetUtilsLibrary::DestroyWidget(*ProgressionMenuOverlayWidgetInternal);
 		ProgressionMenuOverlayWidgetInternal = nullptr;
 	}
+	CurrentGameStateInternal = ECurrentGameState::None;
```

Comment:
Please don't cache CurrentGameState at all, but always take if from AMyGameState::GetCurrentGameState(), as GameState actor should manage lifetime and handle its own leaks, but not HUD component

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/76226acc442de62fba5815cbe12b16facf7da76c